### ISSUE TITLE
[BUG]: Octoprint Settings Page for Plugin freezes after adding second Received Hook

### ISSUE THREAD

**Issue by chhorisberger** [RoboMagus/OctoPrint-Custom_gcode_events#3]
**Describe the bug**
After adding more than one row in the "Gcode Recevieved Hooks" section, the interface freezes and throws a javascript error.
Every time I click on the "plus" sign, another JS error appears in the developer console
This basically prevents me to use the plugin at all for my purposes :/

**Expected behavior**
It should be possible to add more than one row in the "gcode recevieved hooks" section

**Debug logs**

**Screenshots**

![image](https://github.com/RoboMagus/OctoPrint-Custom_gcode_events/assets/39458896/d5cb4257-30d1-4e31-af5b-5a00208c8c2e)

**Additional context**

First of all, thank you for your work and for publishing this plugin 👍  I really hope there is a way to get this working for me, because this would be a huge time server to be notified if my Prusa MMU is acting up again ;)

But first, more information about the issue I observed

I tried multiple times, with different settings

* With empty ReFireThreshold
* With non-empty RefireThreshold
* With different g-codes, tops and events
* with GCode Sent Hooks disabled or enabled

After clearing plugin data in octoprint plugin manager I was able to add a first row again, but when adding a second one, the error occurs again.

Javascript error message:

packed_libs.js?6bee1dee:129 Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at Object.Ua (packed_libs.js?6bee1dee:129:3719)
    at a.a.ec (packed_libs.js?6bee1dee:134:3802)
    at Object.G (packed_libs.js?6bee1dee:130:11201)
    at x (packed_libs.js?6bee1dee:133:18324)
    at a.$.l (packed_libs.js?6bee1dee:134:525)
    at Function.yd (packed_libs.js?6bee1dee:130:18286)
    at Function.zd (packed_libs.js?6bee1dee:130:17972)
    at Function.ha (packed_libs.js?6bee1dee:130:17808)
    at Function.Hc (packed_libs.js?6bee1dee:130:17653)
    at Function.notifySubscribers (packed_libs.js?6bee1dee:130:9667)

OctoPrint 1.9.3 
Python 3.7.3 
OctoPi 0.18.0
Running on a Raspberry Pi 4
Firefox and Chrome (latest version)

**Comment by RoboMagus** [comment]
Thanks for reporting this issue. 

After a quick test it does seem to replicate on my setup as well, but I'll have to dig deeper to see what's causing the issue.

In the meantime you _could_ make the changes in your `config.yaml` file directly if you're comfortable with that. (Look for the `custom_gcode_events` section). 
As long as you strictly adhere to the same format (see the YAML snippet below that's part of my config) when writing directly to the config file, your changes should be picked up by Octoprint after a restart.

<details>
  <summary>Yaml config snippet</summary>
  
```yaml

  custom_gcode_events:
    received_gcode_hooks:
    - enabled: true
      event: Filament change
      exactMatch: false
      gcode: M600
      reFireThreshold: ''
      topic: action_required
    - enabled: true
      event: Paused for user rcvd
      exactMatch: 'true'
      gcode: 'echo:busy: paused for user'
      reFireThreshold: 120
      topic: action_required
    - enabled: true
      event: thermal runaway
      exactMatch: false
      gcode: THERMAL RUNAWAY
      reFireThreshold: ''
      topic: error
    - enabled: true
      event: power failure
      exactMatch: false
      gcode: INT4
      reFireThreshold: ''
      topic: error
    - enabled: true
      event: crash detected
      exactMatch: false
      gcode: CRASH_DETECTED
      reFireThreshold: ''
      topic: error
    sent_gcode_hooks:
    - enabled: true
      event: Paused for user sent
      exactMatch: false
      gcode: M601
      reFireThreshold: ''
      topic: action_required
    - enabled: true
      event: Emergency Stop
      exactMatch: 'true'
      gcode: M112
      reFireThreshold: ''
      topic: fatal
    - enabled: true
      event: Quick Stop
      exactMatch: 'true'
      gcode: M410
      reFireThreshold: ''
      topic: fatal
    - enabled: true
      event: Break
      exactMatch: 'true'
      gcode: M108
      reFireThreshold: ''
      topic: fatal

```

</details>

**Comment by chhorisberger** [comment]
Thanks for the quick reply and the suggested workaround! I'll try to add the hooks by editing the config.yaml until the issue is resolved 👍

**Comment by RoboMagus** [comment]
Turns out I created this bug November last year with the addition of the addition of the configurable re-fire threshold.
It's fixed now in release [0.3.2](https://github.com/RoboMagus/OctoPrint-Custom_gcode_events/releases/tag/0.3.2).

**Comment by chhorisberger** [comment]
Just upgraded to 0.3.2 and can confirm the issue is fixed now. thanks again!
